server/common/fbs/matrix.py

```diff
@@ -46,74 +50,91 @@ def serialize_matrix(builder, n_rows, n_cols, columns, col_idx):
 # Serialization helper
 def serialize_typed_array(builder, source_array, encoding_info):
     """
-    Serialize any of the various typed arrays, eg, Float32Array. Specific means of serialization and type conversion
+    Serialize any of the various typed arrays, eg, Float32FBArray. Specific means of serialization and type conversion
     are provided by type_info.
     """
 
     arr = source_array
-    (array_type, as_type) = encoding_info(source_array)
-
+    (array_type, as_type1, as_type2) = encoding_info(source_array)
     if isinstance(arr, pd.Index):
         arr = arr.to_series()
 
-    # convert to a simple ndarray
-    if as_type == "json":
+    types = [as_type1, as_type2, "int"]
```

Comment:
this hard-wires the assumption that the third slot is _always_ an int.  Ie, that is a very hidden shared dependency between the matrix.fbs file and this code.  Seem incredibly error prone - any change to the FBS file will break this.

I suggest you either:
1. add the third slot to the encoding table
2. comment the heck out of this in _both_ files

I prefer the first, but that is just an opinion. Even the encoding table is a shared assumption between the two files (just trying to put all of the assumptions in one place)

Reply:
Great suggestion. I'm on it.